mixer/pkg/perf/clientserver.go

```diff
@@ -92,15 +92,20 @@ func (s *ClientServer) initializeRPCServer() error {
 		return err
 	}
 
-	s.rpcPath = generatePath("client", s.listener.Addr())
-	rpcDebugPath := generateDebugPath("client", s.listener.Addr())
+	s.rpcPath = generatePath("client")
+	rpcDebugPath := generateDebugPath("client")
 
 	s.rpcServer = rpc.NewServer()
 	_ = s.rpcServer.Register(s)
 	s.rpcServer.HandleHTTP(s.rpcPath, rpcDebugPath)
 
 	go func() {
-		_ = http.Serve(s.listener, nil)
+		// Make a local copy of the listener before using. Since this function closes over s, it is possible
+		// that somebody might have called s.close() before this goroutine starts running.
+		l := s.listener
```

Comment:
you could capture it above, before the go routine (and may need a lock to check the value - did you check what happens if calling Serve on a non nil but closed listener ?)

Reply:
Not sure why capturing "l" on the outside would be better? 
Read the code for serving path, the server will return an error through its loop if the Accept fails. Listener is multi-thread safe (according to its documentation). I checked some implementations of Listener and they seem to return an error (not panic) when there is a listen error. This dovetails nicely with http.Serve semantics, as it always returns an error according to its documentation.
https://golang.org/src/net/http/server.go


The listener will return errOnClose

Reply:
Will update to capture locally.